### ISSUE TITLE
Having a pyproject.toml will break diagnosticSeverityOverrides in settings.json

### ISSUE THREAD

**Issue by maflAT** [microsoft/pyright#8206]
Diagnostic severity overrides in user settings or workspace settings are ignored if the workspace also contains a `pyproject.toml`.

## Environment data
Pylance v2024.6.101

## Repro Steps

1. open an empty folder in VSCode
2. create a `.vscode/settings.json` file with the following contents
    ```json
    {
        "python.analysis.diagnosticSeverityOverrides": {
            "reportUnknownParameterType": "warning"
        }
    }
    ```
3. create a `test.py` file with contents:
    ```python
    def test(x): ...
    ```
4. create an empty `pyproject.toml` file

## Expected behavior

![grafik](https://github.com/microsoft/pylance-release/assets/13979550/74a87ba4-6aad-452b-936d-f2a5881585b3)

## Actual behavior
No error or warning is reported.
![grafik](https://github.com/microsoft/pylance-release/assets/13979550/d5943149-7b43-4278-af79-88325d45db6b)

Deleting `pyproject.toml` will fix the issue.

The same behavior applies to user settings.
This issue is not present in the current release version of pylance (2024.6.1).

**Comment by erictraut** [comment]
It's possible this is related to [this commit](https://github.com/microsoft/pyright/commit/e588c6f984cdca108178c869329974bd49137a2b). I tested that diagnosticSeverityOverrides still work in pyright, but maybe there's an issue that affects pylance's logic that layers onto pyright.

Could someone from the pylance team please take this one?
